src/lib/support/tests/TestCHIPMemString.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements a unit test suite for CHIP Memory Management
+ *      code functionality.
+ *
+ */
+
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <lib/support/CHIPMemString.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/ScopedBuffer.h>
+#include <lib/support/Span.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <nlunit-test.h>
+
+using namespace chip;
+using namespace chip::Platform;
+
+// =================================
+//      Unit tests
+// =================================
+
+namespace {
+template <size_t testStrLen>
+struct TestBuffers
+{
+    char correctSizeBuf[testStrLen + 1];
+    char tooSmallBuf[testStrLen];
+    char wayTooSmallBuf[1];
+    char tooBigBuf[testStrLen + 10];
+    void Reset()
+    {
+        memset(correctSizeBuf, 0, sizeof(correctSizeBuf));
```

Comment:
How about resetting to non-zero? So that any checks for null terminator validate that 'null terminator explicitly added' rather than already being null in the first place.

Reply:
done.